### PR TITLE
Remove Traffic Ops API version 2

### LINKED ISSUE #6697: Remove Traffic Ops APIv2
## This Improvement request (usability, performance, tech debt, etc.) affects these Traffic Control components:
- Traffic Ops

## Current behavior:
The deprecated Traffic Ops API version 2 exists.

## New behavior:
The deprecated Traffic Ops API version 2 no longer exists.

### PR DESCRIPTION
**Pull request by zrhoffman**
This PR closes #6697 by removing Traffic Ops API v2, logic relating to API v2, and the v2 client and API tests.
<hr/>

## Which Traffic Control components are affected by this PR?

- Traffic Ops
- Traffic Ops
- Traffic Ops Client - (Go library, Python)
- CDN-in-a-Box
- Automation - GitHub Actions

## What is the best way to verify this PR?

Request TO API v2 routes, verify you get HTTP status code 501 Not Implemented:

```json
{
  "alerts": [
    {
      "level": "error",
      "text": "The requested api version is not implemented by this server. If you are using a newer client with an older server, you will need to use an older client version or upgrade your server."
    }
  ]
}
```

## PR submission checklist
- [x] This PR removes tests 
- [x] API v2 documentation was already removed in #6783 
- [x] This PR has a CHANGELOG.md entry 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

**Comment by ocket8888** [comment]
It appears that the `/deliveryservices` endpoint thinks you removed 3.0 as well:

>
```
2022-07-07T22:54:17.999425716Z ERROR: api.go:263: 2022-07-07T22:54:17.7269333Z: 172.18.0.2:40418 TODeliveryService.Read called with invalid API version: 3.0
```

File: traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go [hunk 1]
```diff
@@ -608,43 +580,15 @@ func (ds *TODeliveryService) Read(h http.Header, useIMS bool) ([]interface{}, er
 		// NOTE: it's required to handle minor version cases in a descending >= manner
 		case version.Major > 3:
 			returnable = append(returnable, ds.RemoveLD1AndLD2())
-		case version.Major > 2 && version.Minor >= 1:
+		case version.Major == 3 && version.Minor >= 1:
 			returnable = append(returnable, ds.DowngradeToV31())
-		case version.Major > 2:
-			returnable = append(returnable, ds.DowngradeToV31().DeliveryServiceV30)
```
Comment (rawlinp) [inline traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go:614]: I think this part may be important for differentiating v3.0 and v3.1

File: traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go [hunk 2]
```diff
@@ -128,34 +128,6 @@ func GetDSTLSVersions(dsID int, tx *sql.Tx) ([]string, error) {
...
-	}
-	defer inf.Close()
-
-	ds := tc.DeliveryServiceNullableV15{}
-	if err := json.NewDecoder(r.Body).Decode(&ds); err != nil {
-		api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, errors.New("decoding: "+err.Error()), nil)
-		return
-	}
-
-	res, status, userErr, sysErr := createV15(w, r, inf, ds)
```
Comment (rawlinp) [inline traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go:152]: I believe you can also remove the `createV15` function now too

File: traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go [hunk 3]
```diff
@@ -608,43 +580,15 @@ func (ds *TODeliveryService) Read(h http.Header, useIMS bool) ([]interface{}, er
...
-	id := inf.IntParams["id"]
-
-	ds := tc.DeliveryServiceNullableV15{}
-	if err := json.NewDecoder(r.Body).Decode(&ds); err != nil {
-		api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, errors.New("malformed JSON: "+err.Error()), nil)
-		return
-	}
-	ds.ID = &id
-
-	res, status, userErr, sysErr := updateV15(w, r, inf, &ds)
```
Comment (rawlinp) [inline traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go:641]: I believe you can also remove the `updateV15` function as well

File: traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go [hunk 4]
```diff
@@ -128,34 +128,6 @@ func GetDSTLSVersions(dsID int, tx *sql.Tx) ([]string, error) {
...
 }
 
-// CreateV15 is the POST handler for APIv2's deliveryservices endpoint, named
-// with "V15" for legacy reasons.
-// TODO allow users to post names (type, cdn, etc) and get the IDs from the
-// names. This isn't trivial to do in a single query, without dynamically
-// building the entire insert query, and ideally inserting would be one query.
-// But it'd be much more convenient for users. Alternatively, remove IDs from
-// the database entirely and use real candidate keys.
-func CreateV15(w http.ResponseWriter, r *http.Request) {
```
Comment (rawlinp) [inline traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go:138]: I see Create and Update being removed, but I believe there is also a switch case in the `Read` method which checks for version < 3

**Review by rawlinp — CHANGES_REQUESTED** [review]

File: traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go (hunk 1 above)
Comment (zrhoffman) [reply traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go:614]: Oops! Re-added but with a `3` in 9e4ad30373.

**Comment by zrhoffman** [comment]
> It appears that the `/deliveryservices` endpoint thinks you removed 3.0 as well:
> 
> > 
> 
> ```
> 2022-07-07T22:54:17.999425716Z ERROR: api.go:263: 2022-07-07T22:54:17.7269333Z: 172.18.0.2:40418 TODeliveryService.Read called with invalid API version: 3.0
> ```

Fixed in 9e4ad30373

File: infrastructure/cdn-in-a-box/traffic_ops_integration_test/Dockerfile [hunk 5]
```diff
@@ -56,7 +55,7 @@ RUN go mod vendor -v
 
 ENV CGO_ENABLED=0
 RUN set -o errexit -o xtrace;\
-    for api_version in v2 v3 v4; do\
+    for api_version in v3 v4; do\
         go test -c ./$api_version -ldflags="-w -s" -o traffic_ops_${api_version}_integration_test;\
     done
```
[the remark this answers is not in the document]
Comment (zrhoffman) [reply infrastructure/cdn-in-a-box/traffic_ops_integration_test/Dockerfile:61]: Thanks, removed in 2f7626f611

File: traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go (hunk 3 above)
Comment (zrhoffman) [reply traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go:641]: Removed `updateV15` in 55cb74b94c

File: traffic_ops/traffic_ops_golang/server/servers.go [hunk 6]
```diff
@@ -908,14 +778,12 @@ func getServers(h http.Header, params map[string]string, tx *sqlx.Tx, user *auth
 		"dsId":             {Column: "dss.deliveryservice", Checker: nil},
 	}
 
-	if version.Major >= 3 {
-		queryParamsToSQLCols["cachegroupName"] = dbhelpers.WhereColumnInfo{
-			Column:  "cg.name",
-			Checker: nil,
-		}
+	queryParamsToSQLCols["cachegroupName"] = dbhelpers.WhereColumnInfo{
```
Comment (rawlinp) [inline traffic_ops/traffic_ops_golang/server/servers.go:781]: nit: this could be moved up with the rest of the entries

Comment (zrhoffman) [reply traffic_ops/traffic_ops_golang/server/servers.go]: Entry moved up in e1534cc5a6

**Review by rawlinp — APPROVED** [review]
All my comments have been addressed, but I'll wait for the other reviewers to approve as well.

File: traffic_ops/traffic_ops_golang/dbhelpers/db_helpers.go [hunk 7]
```diff
@@ -2123,8 +2123,8 @@ func UpdateServerProfilesForV4(id int, profile []string, tx *sql.Tx) error {
 	return nil
 }
 
-// UpdateServerProfileTableForV2V3 updates CommonServerPropertiesV40 struct and server_profile table via Update (server) function for API v2/v3.
-func UpdateServerProfileTableForV2V3(id *int, newProfileId *int, origProfile string, tx *sql.Tx) error {
+// UpdateServerProfileTableForV3 updates CommonServerPropertiesV40 struct and server_profile table via Update (server) function for API v3.
+func UpdateServerProfileTableForV3(id *int, newProfileId *int, origProfile string, tx *sql.Tx) error {
```
Comment (ericholguin) [inline traffic_ops/traffic_ops_golang/dbhelpers/db_helpers.go:2127]: nit: Comment below also references V2. Line 2141

**Review by ericholguin — COMMENTED** [review]
Can the traffic_ops/testing/api//v2 folder be removed as well, I'm still seeing it

**Comment by zrhoffman** [comment]
> Can the traffic_ops/testing/api//v2 folder be removed as well, I'm still seeing it

folder gitignore removed in a712983cab

File: traffic_ops/traffic_ops_golang/dbhelpers/db_helpers.go (hunk 7 above)
Comment (zrhoffman) [reply traffic_ops/traffic_ops_golang/dbhelpers/db_helpers.go:2127]: Removed reference in fc419ac2ad

**Review by ericholguin — APPROVED** [review]

**Review by ocket8888 — APPROVED** [review]
my concerns were addressed

**Pull request merged**